### PR TITLE
Fix the bundle when soft deps aren't installed

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

This PR fixes some edge cases. It's require to be compatible with Symfony Flex.

### CODE REVIEW

File: src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php [hunk 1]
```diff
@@ -131,8 +131,11 @@ private function createMetadata(ApiProperty $annotation, PropertyMetadata $paren
 
     private function createWith(PropertyMetadata $propertyMetadata, array $property, $value): PropertyMetadata
     {
-        $getter = $property[0].ucfirst($property[1]);
+        if (null === $value) {
+            return $propertyMetadata;
+        }
```
Comment (soyuka) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: Why this change?

File: composer.json [hunk 2]
```diff
@@ -46,12 +46,12 @@
...
-        "symfony/phpunit-bridge": "^2.7 || ^3.0",
-        "symfony/security": "^2.7 || ^3.0",
-        "symfony/validator": "^2.7 || ^3.0",
         "symfony/finder": "^2.7 || ^3.0",
         "symfony/framework-bundle": "^3.1",
-        "symfony/twig-bundle": "^2.8 || ^3.1"
+        "symfony/phpunit-bridge": "^2.7 || ^3.0",
+        "symfony/security": "^2.7 || ^3.0",
+        "symfony/twig-bundle": "^2.8 || ^3.1",
+        "symfony/validator": "^2.7 || ^3.0"
```
Comment (dunglas) [inline composer.json]: It is in alphabetical order isn't it?

File: src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php (hunk 1 above)
Comment (dunglas) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: Because it was leading to a bug if the value is null (the default)

File: composer.json (hunk 2 above)
Comment (soyuka) [inline composer.json]: :laughing: omg friday you know (and I've apparently issues with the alphabet).

File: src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php (hunk 1 above)
Comment (soyuka) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: Okay, I'd have kept in from calling this method in the above loop instead of adding the condition here. Weird that we had no issues about this though oO.

Comment (dunglas) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: Done (it removes some function calls in a loop, it's always appreciable).

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 3]
```diff
@@ -129,16 +134,71 @@ private function handleConfig(ContainerBuilder $container, array $config, array
...
+
+        if (class_exists(Yaml::class)) {
+            $loader->load('metadata/yaml.xml');
+            $container->getDefinition('api_platform.metadata.extractor.yaml')->addArgument($yamlResources);
+        }
 
-        if (!interface_exists(DocBlockFactoryInterface::class)) {
-            $container->removeDefinition('api_platform.metadata.resource.metadata_factory.php_doc');
+        if (interface_exists(DocBlockFactoryInterface::class)) {
+            $loader->load('metadata/php_doc.xml');
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: Why `php_doc` instead of `phpdoc`? For example the filename of the PSR draft is https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md

Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: Just like PHPUnit is always `phpunit`, not `php_unit` or `php-unit` etc.
